### ISSUE TITLE
Capitalized File Names

### ISSUE THREAD

**Issue by psaunders** [turbulent/substance#17]
This issue causes me problems a handful of times: 
https://github.com/EugenMayer/docker-sync/issues/221

It happens rarely enough that I forget about it and waste half an hour. Comes up a lot reorganising files to follow PSR4 autoloading specs.

Solutions?
- Message in logs
- Another layer of checking (like a job that runs every minute to check/fix discrepancies)
- A better warning in the software being developed that a file doesn't exist

The immediate fix is to totally change the filename, wait a few seconds (edit some content) then change it back with the capitalization correct.

**Comment by bbeausej** [comment]
Yes, I ran into this a few times too. Checking for discrepancies might be difficult since we would have to read the unison archive files to follow the synchronizer state and basically check the capitalization.

Worth more research.

**Comment by Kiniamaro** [comment]
is this a mac only issue? reading EugenMayer/docker-sync#221, it seems it's due to HFS+ being trash.


>**EugenMayer:** Sorry but this is unfixable, that is exactly what @rodrigoaguilera supposed. Its like the same issue with git under OSX

> You can fix it by using case sensitive HFS partition, but be sure that you do not do this for the root partition..or you will have hard times.


seems like this is more a problem with unison than substance. might be worth opening an issue there if one isn't up already.
